Zotero/Extensions/MD5+Url.swift

```diff
@@ -6,41 +6,19 @@
 //  Copyright © 2019 Corporation for Digital Scholarship. All rights reserved.
 //
 
-import CommonCrypto
+import CryptoKit
 import Foundation
 
 import CocoaLumberjackSwift
 
 func md5(from url: URL) -> String? {
-    let bufferSize = 1024 * 1024
 
     do {
-        // Open file for reading:
-        let file = try FileHandle(forReadingFrom: url)
-        defer {
-            file.closeFile()
-        }
-
-        // Create and initialize MD5 context:
-        var context = CC_MD5_CTX()
-        CC_MD5_Init(&context)
-
-        // Read up to `bufferSize` bytes, until EOF is reached, and update MD5 context:
-        while autoreleasepool(invoking: {
-            let data = file.readData(ofLength: bufferSize)
-            if data.count > 0 {
-                data.withUnsafeBytes {
-                    _ = CC_MD5_Update(&context, $0.baseAddress, numericCast(data.count))
-                }
-                return true // Continue
-            } else {
-                return false // End of file
-            }
-        }) { }
+        // Read data from file URL
+        let fileData = try Data(contentsOf: url)
```

Comment:
You simplified it a bit too much. In this case you load the whole file to memory. Since we're calculating MD5 hash from files which can potentially be very big and not fit into memory, you need to use another approach here: https://developer.apple.com/documentation/cryptokit/insecure/md5/3237554-update.

Reply:
I did use the update method as well but as far as I understand the orignal code, the file was read into a buffer in 1 MB chunks and the hash was updated accordingly until it reached EOF. In the end we read the same amount of bytes.

If the issue is a potential blocking of the thread, we can probably dispatch an async thread or indeed do the update and finalise. What would you prefer?